### PR TITLE
PUBDEV-6364: Add support for saving model to S3 directly (no s3a/s3n needed)

### PR DESCRIPTION
**Pull request by michalkurka**

### CODE REVIEW

File: h2o-persist-s3/src/main/java/water/persist/PersistS3.java [hunk 1]
```diff
@@ -169,8 +150,77 @@ public InputStream open(String path) {
...
+            LOG.warn("Temporary file `" + _file.getAbsolutePath() + "` was not deleted. Please delete manually.");
+          }
+        }
+      }
+    }
+  } 
+
+  static class CallbackFileOutputStream extends FileOutputStream {
+    private final Object closeLock = new Object();
+    private volatile boolean closed = false;
```
Comment (koniecsveta) [inline h2o-persist-s3/src/main/java/water/persist/PersistS3.java:205]: `volatile` is redundant when accessing only from synchronized block

Comment (michalkurka) [reply h2o-persist-s3/src/main/java/water/persist/PersistS3.java:205]: good catch, this is just copy&paste from FileOutputStream; they are doing the same thing - I can remove it but I would rather keep it identical

@koniecsveta WDYT?

Comment (koniecsveta) [reply h2o-persist-s3/src/main/java/water/persist/PersistS3.java:205]: :+1:

**Comment by koniecsveta** [comment]
LGTM, Thanks!

**Review by koniecsveta — APPROVED** [review]

**Comment by michalkurka** [comment]
thank you for the review @koniecsveta

**Pull request merged**